simulator/kernel/mars/mars_kernel.cpp

```diff
@@ -228,9 +232,26 @@ void MARSKernel::read_from_file() {
     mem->memcpy_host_to_guest( buffer_ptr, byte_cast( buffer.data()), chars_to_read);
 }
 
-void MARSKernel::connect_memory( std::shared_ptr<FuncMemory> m)
+void MARSKernel::connect_riscv_handler()
+{
+    constexpr Addr TRAP_VECTOR = 0x8'000'0000;
+    sim->write_csr_register( "stvec", TRAP_VECTOR);
+    auto pc = trap_vector_address<Addr>(TRAP_VECTOR, 3);
+    ElfLoader elf_loader( KERNEL_IMAGES "riscv32.bin");
+    elf_loader.load_to( mem.get(), pc - elf_loader.get_text_section_addr());
+}
+
+void MARSKernel::connect_mars_handler()
 {
-    BaseKernel::connect_memory( m);
     ElfLoader elf_loader( KERNEL_IMAGES "mars32_le.bin");
     elf_loader.load_to( mem.get(), 0x8'0000'0180 - elf_loader.get_text_section_addr());
 }
+
+void MARSKernel::connect_exception_handler()
+{
+    auto isa = sim->get_isa();
+    if ( isa == "riscv32")
+        connect_riscv_handler();
+    if ( isa == "mars" || isa == "mips32le" || isa == "mips32")
+        connect_mars_handler();
```

Comment:
What happens if other ISA is provided?

Reply:
Nothing

Reply:
And what should happen? :-)

Reply:
THe answer is to throw an exception, since user's input is incorrect: he or she wants to use MARS kernel on a not supported ISA. Please add a new type of exception and write a small test.